File: frontend/src/types/llm.ts
export type LlmLanguage = "ru" | "en";

export interface CategorizeRequest {
  title: string;
  description?: string | null;
  language: LlmLanguage;
}

export interface CategorizeResponse {
  category: string;
}

export interface DecomposeRequest {
  title: string;
  description?: string | null;
  language: LlmLanguage;
}

export interface SubtaskSuggestion {
  title: string;
  description: string | null;
}

export interface DecomposeResponse {
  subtasks: SubtaskSuggestion[];
}

export interface WorkloadSummaryResponse {
  summary: string;
  task_count: number;
  overdue_count: number;
}

File: backend/app/schemas/llm.py
from enum import StrEnum

from pydantic import BaseModel, Field


class Language(StrEnum):
    RU = "ru"
    EN = "en"


class CategorizeRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=10_000)
    language: Language = Language.RU


class CategorizeResponse(BaseModel):
    category: str = Field(..., min_length=1, max_length=64)


class DecomposeRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=10_000)
    language: Language = Language.RU


class SubtaskSuggestion(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=10_000)


class DecomposeResponse(BaseModel):
    subtasks: list[SubtaskSuggestion] = Field(..., min_length=1, max_length=10)


class WorkloadSummaryResponse(BaseModel):
    summary: str
    task_count: int
    overdue_count: int
